src/realtime_metrics/main.py

```diff
@@ -175,25 +185,151 @@ def eta_accuracy(stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]):
     # compute total accuracy
     if len(accuracies) == 0:
         logger.info("No data provided!")
-        return
+        return 0.0
     mean_accuracy = numpy.mean(accuracies) * 100
     return mean_accuracy
 
 
+def get_actual_delay(trip_update: TripUpdate, stop_time_update: StopTimeUpdate) -> int | None:
+    """
+    Returns the actual delay for the route and stop of the given TripUpdate.
+    If no actual delay is known, None is returned.
+    """
+    key = (trip_update.route_id, trip_update.trip_id, stop_time_update.stop_id)
+    if key in actual_arrival_times.keys():
+        newest_stop_time_update: StopTimeUpdate = actual_arrival_times[key][1]
+        return newest_stop_time_update.arrival_delay
+    else:
+        return None
+
+
+def get_actual_arrival_time(trip_update: TripUpdate, stop_time_update: StopTimeUpdate) -> int | None:
+    """
+    Returns the actual arrival time for the route and stop of the given TripUpdate.
+    If no actual arrival time is known, None is returned.
+    """
+    key = (trip_update.route_id, trip_update.trip_id, stop_time_update.stop_id)
+    if key in actual_arrival_times.keys():
+        newest_stop_time_update: StopTimeUpdate = actual_arrival_times[key][1]
+        return newest_stop_time_update.arrival_time
+    else:
+        return None
+
+
+def experienced_wait_time_delay(trip_stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]) -> float:
```

Comment:
Could also return None
```suggestion
def experienced_wait_time_delay(trip_stop_time_updates: list[tuple[TripUpdate, StopTimeUpdate]]) -> floating[
                                                                                                        Any] | None:
```